src/metabase/models/database.clj

```diff
@@ -299,12 +299,16 @@
                                      details
                                      (sensitive-fields-for-db db)))))]
      (update db :settings (fn [settings]
-                            (when settings
+                            (when (map? settings)
                               (into {}
                                     (filter (fn [[setting-name _v]]
-                                              (setting/can-read-setting? setting-name
-                                                                         (setting/current-user-readable-visibilities))))
-                                    settings)))))
+                                              (try
+                                               (setting/can-read-setting? setting-name
```

Comment:
so here lies the bug. it was very hard to debug because this fails silently!

even though `(setting/can-read-setting? setting-name)` throw an error, the `to-json` method ignores it.

And why it throws an error? after https://github.com/metabase/metabase/pull/31755 I migrated `:persist-models-enabled` into `settings`.

But I didn't know that we need to do a `defsetting` for anything that goes into `metabase_database.settings` 😢.

So this throws an error => `to-json` fails silently => one Tuesday's night of my is gone.

Reply:
anw, let's ignore the exception for now and assume any settings that are not readable will be visible.